content/blog/2020-03-dplyr-1-0-0-new-verb-features/index.Rmarkdown

```diff
@@ -0,0 +1,172 @@
+}
+
+df %>% 
+  group_by(grp) %>% 
+  summarise(quibble2(y, c(0.25, 0.5, 0.75)))
+```
+
+One note of caution: name the output columns in a function like this is a surprisingly complex task, we're not yet sure what the best approach expect. Expect to here more about this as we continue to think about and experiment with it.
+
+## Data frame columns
+
+We've been careful not to name the result of `quibble()` in the code above. That's because we leave the name off, the data frame result is automatically **unpacked** so each column returned by `quibble()` becomes a column in the result. What happens if we name the output?
+
+```{r}
+out <- df %>% 
+  group_by(grp) %>% 
+  summarise(y = quibble2(y, c(0.25, 0.75)))
+out
+```
+Look carefully at the output - you'll see a `$` in the column names. This lets you know that something weird is going on and you have what we call a **df-column**; a column of a data frame that is itself a data frame! 
+
+You can see the structure a little better with `str()`:
+
+```{r}
+str(out)
+```
+
+And you can that `y` is indeed a data frame by extracting it:
+
+```{r}
+out$y
+```
+
+And of course, you can dig still deeper to get the individual values:
+
+```{r}
+out$y$y
+```
+
+These df-columns are simultaneously esoteric and common place. On the one hand they are an oddity of data frames that has existed for a long time, but has been used in very few places. On the other hand, they are very closely related to merged column headers, which, judging by how often they're found in spreadsheets, are an incredibly popular tool. Our hope is that they are mostly kept under the covers in dplyr 1.0.0, but you can still deliberately choose to access them if you're interested.
+
+## Non-summaries
+
+In combination with [`rowwise()`](http://dplyr.tidyverse.org/dev/articles/rowwise.html) (more on that in a future blog post), `summarise()` is now sufficiently powerful to replace many workflows that previously required a `map()` or `apply()` function. 
+
+For example, to read all the all the `.csv` files in the current directory, you could write:
+
+```{r, eval = FALSE}
+tibble(path = dir(pattern = "\\.csv$")) %>% 
+  rowwise(path) %>% 
+  summarise(read_csv(path))
+```
+
+I feel deeply ambivalent about this code: it seems rather forced to claim that `read.csv()` computes a summary of a file path, but it's rather elegant pattern for reading in many files into a tibble.
+
+## Previous approaches
+
+There were a couple of previous approach to solving the quantile problem illustrated above. One way was to create a list-column and then unnest it:
+
+```{r}
+df %>% 
+  group_by(grp) %>% 
+  summarise(y = list(quibble(y, c(0.25, 0.75)))) %>% 
+  tidyr::unnest(y)
+```
+
+Or to use `do()`:
+
+```{r}
+df %>% 
+  group_by(grp) %>% 
+  do(quibble(.$y, c(0.25, 0.75)))
+```
+
+We prefer the new `summarise()` approach with because it's concise, doesn't require learning about list-columns and unnesting, and uses a familiar syntax.
```

Comment:
```suggestion
We prefer the new `summarise()` approach because it's concise, doesn't require learning about list-columns and unnesting, and uses a familiar syntax.
```